package/kernel/linux/modules/netdevices.mk

```diff
@@ -642,6 +642,39 @@ endef
 $(eval $(call KernelPackage,ixgbevf))
 
 
+define KernelPackage/i40e
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Intel(R) Ethernet Controller XL710 Family support
+  DEPENDS:=@PCI_SUPPORT +kmod-mdio +kmod-ptp +kmod-hwmon-core
+  KCONFIG:=CONFIG_I40E \
+    CONFIG_I40E_DCB=n
+  FILES:=$(LINUX_DIR)/drivers/net/ethernet/intel/i40e/i40e.ko
+  AUTOLOAD:=$(call AutoLoad,36,i40e)
```

Comment:
Is there a reason for AutoLoad? Because AutoProbe is definitely preferred because it only loads the driver if there's really a device. 

(The same goes for the i40evf)

Reply:
I40e is Physical Functions drivers
i40evf is Virtual Functions drivers
Replace AutoLoad with AutoProbe @ 474061cf82

Reply:
You don't need to replace the autoload for devices you don't have and can't test. Please focus on your i40e+vf instead.  It would be great if you could squash the [changes into one commit](https://openwrt.org/submitting-patches#squashing_commits).

Also, please add your github PR's text into the patch commit message as well.
(reason being: only the patches will be merged, so the extensive tested on
list will be lost)

Reply:
OK